### PR TITLE
fix: (Core) Switch accessibility

### PR DESCRIPTION
**Pull request by Betrozov**
#### Please provide a link to the associated issue.
https://github.com/sap/fundamental-ngx/issues/3495
https://github.com/sap/fundamental-ngx/issues/3496
#### Please provide a brief summary of this pull request.
This PR contains fixes for the accessibility bugs for Switch component.
#### Please check whether the PR fulfills the following requirements

- [X] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [X] tests for the changes that have been done
- [X] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [X] update `README.md`
- [X] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [X] Documentation Examples
- [X] Stackblitz works for all examples

### CODE REVIEW

**Review by mikerodonnell89 — APPROVED** [review]

File: libs/core/src/lib/switch/switch.component.html [hunk 1]
```diff
@@ -1,40 +1,52 @@
...
+                    class="fd-switch__icon--off fd-switch__icon sap-icon--decline"
+                ></i>
             </div>
+
+            <span
+                *ngIf="semantic"
+                [id]="_semanticLabelId"
+                aria-hidden="true"
+                class="fd-switch__invisible-text"
+                i18n="@@coreSwitch.SemanticLabel"
```
Comment (JKMarkowski) [inline libs/core/src/lib/switch/switch.component.html:46]: Is it decided to use i18n on core as well?

File: libs/core/src/lib/switch/switch.component.ts [hunk 2]
```diff
@@ -89,15 +89,15 @@ export class SwitchComponent implements ControlValueAccessor {
...
     /** @hidden */
-    onChange: any = () => {};
+    onChange: Function = () => {};
 
     /** @hidden */
-    onTouched: any = () => {};
+    onTouched: Function = () => {};
 
-    constructor(private changeDetectorRef: ChangeDetectorRef) {}
+    constructor(private readonly changeDetectorRef: ChangeDetectorRef) {}
```
Comment (JKMarkowski) [inline libs/core/src/lib/switch/switch.component.ts:97]: Could you add `_` prefix to changedetectorRef?

**Review by JKMarkowski — APPROVED** [review]
LGTM

**Comment by AntonOlkhovskyi** [comment]
Tested. Looks good.

**Review by AntonOlkhovskyi — APPROVED** [review]

**Pull request merged**